python/src/evaluate.py

```diff
@@ -38,12 +40,15 @@ class ExactEvalFunction(EvalFunction):
     '''
     Eval Function to do exact matching of the keywords
     '''
+    WORD_MATCHER = r"[a-zA-Z]+"
 
     def pre_process(keywords):
         '''
         Basic processing, lower the keywords
         '''
-        return [word.lower() for word in keywords]
+        lower_keywords = [word.lower() for word in keywords]
+        return [' '.join(re.findall(self.WORD_MATCHER, word))
```

Comment:
This is not filtering out non-alpha chars. Eg:

In [6]: pre_process(['#ree', 'this', 'ThISsd', 'sadJH66sSD'])
Out[6]: ['#ree', 'this', 'thissd', 'sadjh66ssd']